Consistent eval parameters and allowed arguments

Kronuz opened the pull request:
The default value for the eval argument (True) should be consistent across all `__new__()` and `__init__()`. Also, Symbol should only receive one optional positional argument, no more. Added test for that as well.

pombredanne commented on boolean/boolean.py:
```diff
@@ -457,8 +463,9 @@ class Function(Expression):
-        assert len(kwargs) == 0
+        eval = kwargs.pop('eval', True)
+        if kwargs:
+            raise TypeError("Got an unexpected keyword argument '%s'" % kwargs.keys()[0])
```
would `%r` be better than `'%s'`?

Kronuz replied:
Absolutely, I'll change the PR

pombredanne commented on boolean/boolean.py:
```diff
@@ -474,10 +481,11 @@ def __init__(self, *args, **kwargs):
         # overwritten.
+        kwargs.pop('eval', True)
+        if kwargs:
+            raise TypeError("Got an unexpected keyword argument '%s'" % kwargs.keys()[0])
```
Same as above: would `%r` be better than `'%s'`?

pombredanne commented on boolean/boolean.py:
```diff
@@ -363,10 +363,16 @@ class Symbol(Expression):
         return object.__new__(cls)
 
-    def __init__(self, obj=None, eval=False):
+    def __init__(self, obj=None, *args, **kwargs):
```
why do you need variadic args at all here? I reckon this is for consistency, but may be we do not need them at all on any of the initializers?

Kronuz replied:
Just for completeness, so it all can receive `eval`

bastikr commented:
Sorry, I also can't see why this is necessary. All the other things in this patch look fine btw.

Kronuz commented:
@bastikr, what part exactly you think is not necessary? (to remove it from the PR)

bastikr commented:
Using variadic args, i.e.

-    def __init__(self, obj=None, eval=False):
+    def __init__(self, obj=None, *args, **kwargs):

Kronuz commented:
@bastikr, you are right... at the time I thought so so someone couldn't create a symbol like this: Symbol(1, 2) (the 2 being `eval` or `simplify`) ... but I thin you are right. I've removed the variadic args and rebased this PR.

bastikr commented:
Nice! Ready to merge.
